### DISCUSSION TITLE
PyPDF2 2.0: Cleanup plan

### DISCUSSION THREAD

**Discussion by MartinThoma** [py-pdf/pypdf#752] [Ideas]
The PyPDF2 2.X versions should have a clean interface. Here is a list of things I want to do - please comment / add suggestions!

1.  :heavy_check_mark:  **Distribution Package**: Remove Tests, Resources, and Scripts from the package ([discussion](https://github.com/py-pdf/PyPDF2/discussions/718))
2. :heavy_check_mark:  **PEP8 Names**: Use PEP8-compliant function/method/variable names. In many cases that means using snake_case instead of camelCase. There will be a PyPDF2 1.X version that has both + deprecation warnings so that people can make the switch gradually. I would also rename `Tests` to `tests`.
3. **Explicit Exports**: PyPDF2 will explicitly export some modules / classes / functions. Those are meant for users of PyPDF2 to be used. Everything else can be changed without deprecation warnings (discussion: [Which parts of PyPDF2 should be part of the public API?](https://github.com/py-pdf/PyPDF2/discussions/750))
4. **Dropping methods in favor of magic methods**: I have a couple of ideas for a modern interface to PyPDF2 ([links](https://github.com/py-pdf/PyPDF2/discussions/751)). Those can be introduced additionally and don't cause any technical issues having them side-by-side. However, for users it's simpler when the usage of PyPDF2 looks consistent in examples. For this reason I would favor deprecating the non-magic method functions.
5. :heavy_check_mark:  **Dropping overwriteWarnings** ([done](https://github.com/py-pdf/PyPDF2/pull/846)): `warnings.showwarning = _showwarning` cases issues, see #67 - merging https://github.com/py-pdf/PyPDF2/pull/225
6.  :heavy_check_mark:  **Black code formatting**
7. :heavy_check_mark:  **strict=False by default** ([done](https://github.com/py-pdf/PyPDF2/pull/848)): see https://github.com/py-pdf/PyPDF2/discussions/783#discussioncomment-2596336
8. :heavy_check_mark:  **Private methods / functions**: Some methods, like the rectangle accessors mentioned in #517 , should have a leading underscore to mark them as private.
9. :heavy_check_mark: **Remove merger.OutlinesObject** ([done](https://github.com/py-pdf/PyPDF2/pull/847)) : Removing `PyPDF2.merger.OutlinesObject` as nobody seems to use it

PRs that I want to merge with 2.0:

* https://github.com/py-pdf/PyPDF2/pull/749 : Better encryption support

**Comment by MartinThoma** [comment]
Is the `PyPDF2.merger.OutlinesObject` used by anybody? Can somebody show a small script how it's used?

**Comment by sekrause** [comment]
I think auto formatting code with something like https://github.com/psf/black is generally a good idea.

However it's usually not a good idea to simply reformat an existing project because it would break all pull requests, so maybe take a look at https://github.com/akaihola/darker which only auto-formats code lines that have changed in a commit (and maybe only start doing that until all old PRs are either merged or rejected).

**Comment by MartinThoma** [comment]
I'd also deprecate things like `RectangleObject.getLowerRight` in favor of `RectangleObject.lowerRight` (using the properties instead of Java-style getter/setter methods - although things don't change under the hood)
